polymetis/polymetis/src/polymetis_server.cpp

```diff
@@ -294,8 +294,20 @@ Status PolymetisControllerServerImpl::UpdateController(
   if (custom_controller_context_.status == RUNNING) {
     custom_controller_context_.controller_mtx.lock();
     interval->set_start(robot_state_buffer_.size());
-    custom_controller_context_.custom_controller->param_dict_update_module();
+    int code = custom_controller_context_.custom_controller
+                   ->param_dict_update_module();
     custom_controller_context_.controller_mtx.unlock();
+
+    if (code) {
+      std::string error_msg;
+      if (code == 1) {
+        error_msg = "error updating controller: Invalid parameter name.";
+      } else if (code == 2) {
+        error_msg = "error updating controller: Tensor shape mismatch.";
+      }
+      return Status(StatusCode::CANCELLED, error_msg);
+    }
```

Comment:
My thought is that it's hard to exhaustively enumerate all possible exceptions that could possibly happen while calling param_dict_update_module. Instead, can we wrap this in a try/catch, same as e.g. line 237-257, and directly return the error message?

Reply:
These are the most common error messages that users might run into. If we're giving users the freedom to update the controller with any dictionary, I think it would be very helpful to include more descriptive error messages.

Reply:
Yes, but aren't the error messages from catching the exception the same as the ones you're manually writing here?

Reply:
I'll investigate how informative the error messages are if we allow torchscript modules to error out, or if we're able to raise custom exceptions from within torchscript modules.

Right now the torchscript controller module is designed to check for these user input induced errors, ignore the input if something isn't right and return an error code to ensure that the server doesn't crash. So given that torchscript module design enumeration of the error code has to happen somewhere, and I chose to enumerate it outside of the torchscript isolated module so as to get access to c++ stuff such as std strings.

Reply:
I believe the torchscript error just gets propagated as an `std::exception` and you can forward `e.what()` to the user client -- please lmk if that's not the case